### PR TITLE
feat: rename ALS back to CLS

### PR DESCRIPTION
**Pull request by ephys**
## Pull Request Checklist

- [ ] Have you added new tests to prevent regressions?
- [ ] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [x] Did you update the typescript typings accordingly (if applicable)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

Rename ALS back to CLS following this comment: https://github.com/sequelize/sequelize/issues/13741#issuecomment-1329273015

### CODE REVIEW

**Review by WikiRik — COMMENTED** [review]
Don't forget to rename the test file. Rest looks good

File: src/model.d.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ export interface Transactionable {
   /**
    * The transaction in which this query must be run.
    *
-   * If {@link Options.disableAlsTransactions} has not been set to true, and a transaction is running in the current ALS context,
+   * If {@link Options.disableClsTransactions} has not been set to true, and a transaction is running in the current ALS context,
```
Comment (WikiRik) [inline src/model.d.ts:64]: CLS*

File: src/sequelize-typescript.ts [hunk 2]
```diff
@@ -292,7 +292,7 @@ export abstract class SequelizeTypeScript {
       return result;
     };
 
-    const als = this.#transactionAls;
+    const als = this.#transactionCls;
```
Comment (WikiRik) [inline src/sequelize-typescript.ts:295]: cls*

File: test/support.ts [hunk 3]
```diff
@@ -164,7 +164,7 @@ export function createSequelizeInstance(options: Options = {}): Sequelize {
     dialectOptions: options.dialectOptions || config.dialectOptions || {},
     minifyAliases: options.minifyAliases || config.minifyAliases,
     // the test suite was written before ALS was turned on by default.
```
Comment (WikiRik) [inline test/support.ts:166]: CLS*

File: test/integration/als.test.ts [hunk 4]
```diff
@@ -51,10 +51,10 @@ describe('AsyncLocalStorage Transactions (ALS)', () => {
       let t2id;
       await Promise.all([
         vars.alsSequelize.transaction(async () => {
-          t1id = vars.alsSequelize.getCurrentAlsTransaction()!.id;
+          t1id = vars.alsSequelize.getCurrentClsTransaction()!.id;
```
Comment (WikiRik) [inline test/integration/als.test.ts:54]: als is still used a few times in this file

File: src/model.d.ts (hunk 1 above)
Comment (WikiRik) [inline src/model.d.ts:64]:
```suggestion
   * If {@link Options.disableClsTransactions} has not been set to true, and a transaction is running in the current CLS context,
```
Or `AsyncLocalStorage`

File: src/sequelize-typescript.ts (hunk 2 above)
Comment (WikiRik) [reply src/sequelize-typescript.ts:295]: This happens a few other times in this file as well

File: test/support.ts (hunk 3 above)
Comment (WikiRik) [inline test/support.ts:166]:
```suggestion
    // the test suite was written before CLS was turned on by default.
```
Or AsyncLocalStorage

**Comment by ephys** [comment]
Thanks :) I think I'll pause this PR until https://github.com/sequelize/sequelize/pull/15431 is merged to prevent a complicated merge conflict

File: src/sequelize-typescript.ts (hunk 2 above)
Comment (ephys) [reply src/sequelize-typescript.ts]: Done

File: test/integration/als.test.ts (hunk 4 above)
Comment (ephys) [reply test/integration/als.test.ts]: Done

File: src/model.d.ts (hunk 1 above)
Comment (ephys) [reply src/model.d.ts]: Done

File: test/support.ts (hunk 3 above)
Comment (ephys) [reply test/support.ts]: Done

**Review by WikiRik — APPROVED** [review]

**Pull request merged**